apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -328,22 +313,25 @@ private static object ConvertToClrTypeImpl(Value wireValue, V1.Type spannerType,
                     default:
                         throw new ArgumentOutOfRangeException();
                 }
-            if (targetClrType == typeof(IList))
+            if (targetClrType == typeof(ArrayList) || targetClrType == typeof(IList))
+            {
                 switch (wireValue.KindCase)
                 {
-                    case Value.KindOneofCase.NullValue:
-                        return null;
+                    case Value.KindOneofCase.NullValue: return null;
                     case Value.KindOneofCase.ListValue:
-                        return wireValue.ListValue.Values.Select(x => x.ConvertToClrType(spannerType.ArrayElementType))
-                            .ToList();
+                        return new ArrayList(wireValue.ListValue.Values
+                            .Select(x => x.ConvertToClrType(spannerType.ArrayElementType))
+                            .ToList());
                     default:
                         throw new InvalidOperationException(
                             $"Invalid Type conversion from {wireValue.KindCase} to {targetClrType.FullName}");
                 }
-            if (targetClrType == typeof(IDictionary))
+            }
+            if (targetClrType == typeof(Hashtable) ||
```

Comment:
true I could treat it the same way as IList and simply create it and cast to IDictionary.